### PR TITLE
Feature/sc 10 charging history

### PR DESCRIPTION
**Pull request by lgaljer20**
Created functionality for every user to get his charging history (his events that has finished). User can get his history that includes: start time of the charging, end time of the charging, on which charger did he charge, how much kWh has he consumed and what card did he use to charge.

Changes made:
1. Created _**EventDTO**_ to represent **_Event_** entity.
2. Created **_EventResponseDTO_** that is responsible for sending response.
3. Created interface for my service **_EventService_** that inherits it. In **_EventService_** created method **_GetUsersChargingHistory_**.
4. Created **_EventController_** with endpoint for getting users history.
5. Created unit tests for **_EventController_** and _**EventService**_.

All previous tests and newly added tests have successfully passed. ✅

### CODE REVIEW

File: SmartCharger.Business/Services/EventService.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                    {
+                        Success = false,
+                        Message = "User not found.",
+                        Error = "User not found."
+                    };
+                }
+                IQueryable<Event> query = _context.Events.Where(e => e.UserId == userId && e.EndTime != null);
+
+
+                if (!string.IsNullOrEmpty(search))
```
Comment (lraknic20) [inline SmartCharger.Business/Services/EventService.cs:32]: Everything looks good, I just think search by first and last name isn't necessary.

**Review by lraknic — APPROVED** [review]

**Review by lraknic — APPROVED** [review]
Everything looks good, I just think search by first and last name isn't necessary.

**Comment by lgaljer20** [comment]
@lraknic20 you are right, I changed it to only search by card's and charger's name which makes sense. Thanks for the review.

**Review by mkajic20 — APPROVED** [review]
Everything is good.

**Review by irosic20 — APPROVED** [review]
Everything looks good. Keep up the good work!

**Pull request merged**